source/non_matching/quadrature_generator.cc

```diff
@@ -1918,6 +2196,61 @@ namespace NonMatching
 
 
 
+  template <int dim>
+  void
+  DiscreteFaceQuadratureGenerator<dim>::generate_fe_q_iso_q1(
+    const BoundingBox<dim> &unit_box,
+    unsigned int            face_index)
+  {
+    const unsigned int n_subdivisions =
+      reference_space_level_set->n_subdivisions();
+
+    const unsigned int dofs_per_line = n_subdivisions + 1;
+
+    this->clear_quadratures();
+
+    const unsigned int coordinate_direction_face   = face_index / 2;
+    const unsigned int coordinate_orientation_face = face_index % 2;
+
+    std::vector<std::array<unsigned int, dim>> all_subcell_indices;
+    all_subcell_indices.reserve(Utilities::pow(n_subdivisions, dim));
+    all_subcell_indices.emplace_back();
+    all_subcell_indices[0][coordinate_direction_face] =
+      coordinate_orientation_face == 0 ? 0 : n_subdivisions - 1;
+    for (unsigned d = 0; d < dim - 1; ++d)
+      {
+        const unsigned int old_size = all_subcell_indices.size();
+        for (unsigned int i = 1; i < n_subdivisions; ++i)
+          {
+            for (unsigned int j = 0; j < old_size; ++j)
+              {
+                std::array<unsigned int, dim> next_entry =
+                  all_subcell_indices[j];
+                next_entry[(coordinate_direction_face + d + 1) % dim] = i;
```

Comment:
I think this is a usual idiom, see e.g. https://github.com/dealii/dealii/blob/97cf66283eb125d212f2ff5eb335f57dd662e48f/source/base/polynomials_raviart_thomas.cc#L175
Regarding the code, please add a comment to say that we choose the other two indices on the face, including wrap around.